Applications/spire/source/book_view/book_view_window.cpp

```diff
@@ -99,9 +99,6 @@ bool BookViewWindow::eventFilter(QObject* watched, QEvent* event) {
 }
 
 void BookViewWindow::set_current(const Security& s) {
-  if(s == m_current_security) {
```

Comment:
You still need this guard but you will have to compare against the ```m_security_widget```'s current security.

Reply:
I have the SecurityWidget doing the check to ensure that the same security isn't input twice; do I still need this even with that?

Reply:
Yes otherwise all of the side-effects kick in like resetting the ```m_bbo_quote_panel```, the ```m_table```, so on so forth...

Reply:
Okay good.